exercise/submission_models.py

```diff
@@ -31,7 +31,7 @@ def get_queryset(self):
     def create_from_post(self, exercise, submitters, request):
         new_submission = Submission.objects.create(
             exercise=exercise,
-            submission_data=query_dict_to_list_of_tuples(request.POST),
+            submission_data=[i for i in query_dict_to_list_of_tuples(request.POST) if not '__aplus__' == i[0]],
```

Comment:
This lines becomes pretty long, so It could be a good idea to use a temporary variable. In addition, `i` is often used to mean `index` (sometimes `item`, like here, but that is maybe a bit more rare). As the value is always tuple of size 2, you could unpack and pack it again. That is questionable, as then there is repacking and then it's unclear if the order of the values change.

```python
well_named_variable = [
  (name, value)
  for (name, value) in query_dict_to_list_of_tuples(request.POST)
  if name != '__aplus__'
]
new_submission = Submission.objects.create(
￼            exercise=exercise,
￼            submission_data=query_dict_to_list_of_tuples(request.POST),
￼            submission_data=well_named_variable,
```

Something note too, is that typically the condiftion is more readable if it's in order `x not something y` instead of `not x someting y`, even though they are logically the same.